SDL/EndcapGeometry.h

```diff
@@ -18,28 +18,41 @@
 #include "HeterogeneousCore/AlpakaInterface/interface/host.h"
 
 namespace SDL {
-  template <typename TDev>
-  class EndcapGeometry {};
+
+  // Only the full one contains alpaka buffers
+  template <typename TDev, bool full>
+  class EndcapGeometry;
+
+  template <>
+  class EndcapGeometry<DevHost, false> {
```

Comment:
isn't this an ODR violation? `EndcapGeometry<DevHost, false>` will appear in all alpaka backend builds.

A separate non-template `EndcapGeometryZZZ` (put something clear in ZZZ) would be more appropriate; it will be in non-alpaka part of LSTCore

Reply:
I think that's not an ODR violation. That's why I added the second template parameter so that for cpu builds it doesn't end up having a duplicate definition. But I agree that a non-template class would be better.